lib/Facebook.js

```diff
@@ -815,10 +816,274 @@ function Facebookbot(configuration) {
         });
     };
 
+    var broadcast_api = {
+        create_message_creative: function(message, cb) {
+            var uri = 'https://' + api_host + '/' + api_version + '/me/message_creatives?access_token=' + configuration.access_token;
 
+            var messageToBroadcast = {};
 
+            if (typeof(message) == 'string') {
+                messageToBroadcast.text = message;
+            } else {
+                messageToBroadcast = message;
+            }
+
+            var body = {
+                'messages' : [messageToBroadcast]
+            };
+
+            request({
+                method: 'POST',
+                json: true,
+                body: body,
+                uri: uri
+            }, function(err, res, body) {
+                if (err) {
+                    facebook_botkit.log('Could not Create a broadcast message');
+                    cb(err);
+                } else {
+                    if (body.error) {
+                        facebook_botkit.log('ERROR in creating a broadcast message call: ', body.error);
+                        cb(body.error);
+                    } else {
+                        facebook_botkit.debug('Successfully create a broadcast message', body);
+                        cb(null, body);
+                    }
+                }
+            });
+        },
+        send: function(message_creative_id, custom_label_id,  cb) {
```

Comment:
Take a look at the sendBroadcast function I created in this [PR](https://github.com/howdyai/botkit/pull/1181/files#diff-c54b974993736486b57fb75303b19213R943) where we test if a `creative_message_id` string is passed, or a fresh message/array of messages. I prefer that personally, saves the user a callback level if they don't need to store the creative_message_id